linkerd/app/inbound/src/policy.rs

```diff
@@ -38,9 +41,11 @@ pub struct ServerUnauthorized {
     server: Arc<Meta>,
 }
 
-pub trait GetPolicy {
-    // Returns the traffic policy configured for the destination address.
-    fn get_policy(&self, dst: OrigDstAddr) -> AllowPolicy;
+/// Returns the traffic policy configured for the destination address.
+pub trait GetPolicy: Clone + Send + Sync + 'static {
+    type Future: Future<Output = Result<AllowPolicy, Error>> + Unpin + Send;
+
+    fn get_policy(&self, target: OrigDstAddr) -> Self::Future;
 }
```

Comment:
We should provide a blanket impl for all `Service<OrigDstAddr>` -- this lets us use middlewares and `svc::mk` etc.